tests/unit/backend/chalice/api_server/test_v1_collection.py

```diff
@@ -427,8 +430,143 @@ def test__list_collection__check_owner(self):
             self.assertNotIn(private_not_owned, ids)
 
-    def test__delete_collection__OK(self):
-        path = furl(path="/dp/v1/collections/test_collection_id")
+
+class TestCollectionDeletion(BaseAuthAPITest, GenerateDataMixin):
+    def test_delete_collection__ok(self):
+        # Generate test collection
+        collection = self.generate_collection(
+            self.session, visibility=CollectionVisibility.PRIVATE.name, owner="test_user_id"
+        )
+        processing_status_1 = {"upload_status": UploadStatus.WAITING, "upload_progress": 0.0}
+        processing_status_2 = {"upload_status": UploadStatus.UPLOADED, "upload_progress": 100.0}
+
+        dataset_1 = self.generate_dataset(self.session, collection=collection, processing_status=processing_status_1)
+        dataset_2 = self.generate_dataset(self.session, collection=collection, processing_status=processing_status_2)
+        headers = {"host": "localhost", "Content-Type": "application/json", "Cookie": get_auth_token(self.app)}
+
+        test_url = furl(path=f"/dp/v1/collections/{collection.id}")
+        test_url.add(query_params=dict(visibility="PRIVATE"))
+        response = self.app.get(test_url.url, headers=headers)
+        response.raise_for_status()
+
+        body = json.loads(response.body)
+        dataset_ids = [dataset["id"] for dataset in body["datasets"]]
+        self.assertIn(dataset_1.id, dataset_ids)
+        self.assertIn(dataset_2.id, dataset_ids)
+
+        # delete collection
+        response = self.app.delete(test_url.url, headers=headers)
+        response.raise_for_status()
+        self.assertEqual(response.status_code, 202)
+
+        # check collection and datasets tombstoned
+        self.session.expire_all()
+        collection = Collection.get_collection(
+            self.session, collection.id, CollectionVisibility.PRIVATE.name, include_tombstones=True
+        )
+
+        self.assertTrue(collection.tombstone)
+        self.assertTrue(dataset_1.tombstone)
+        self.assertTrue(dataset_2.tombstone)
+
+        response = self.app.get(test_url.url, headers=headers)
+        self.assertEqual(response.status_code, 403)
+
+    def test_delete_collection__dataset_not_available(self):
+        # Generate test collection
+        collection = self.generate_collection(
+            self.session, visibility=CollectionVisibility.PRIVATE.name, owner="test_user_id"
+        )
+        processing_status = {"upload_status": UploadStatus.UPLOADED, "upload_progress": 100.0}
+
+        dataset = self.generate_dataset(self.session, collection=collection, processing_status=processing_status)
         headers = {"host": "localhost", "Content-Type": "application/json", "Cookie": get_auth_token(self.app)}
-        response = self.app.delete(path.url, headers)
+        dataset_url = furl(path=f"/dp/v1/datasets/{dataset.id}/status")
+        response = self.app.get(dataset_url.url, headers=headers)
+        self.assertEqual(response.status_code, 200)
```

Comment:
Remove. This is already tested in dataset.

Reply:
see above comment